src/utils.c

```diff
@@ -204,3 +204,93 @@ JSON_Value* joy_utils_open_resource_parson(const char *filename) {
     return value;
 }
 
+/*
+ *
+ * \brief Copy a json-printable string from the source buffer to the
+ *        destination buffer.
+ *
+ * \param buf Destination buffer.
+ * \param buflen Maximum length of the destination buffer.
+ * \param data Source buffer.
+ * \param datalen Length of the source buffer.
+ *
+ */
+void copy_printable_string(char *buf,
+			   unsigned int buflen,
+			   const void *data,
+			   unsigned int datalen) {
+    const char *d = data;
+
+    while (buflen-- && datalen--) {
+	if (!isprint(*d) || *d == '\"' || *d == '\\' || *d <= 0x1f) { /* json constraints */
+	    break;
+	}
+	*buf++ = *d++;
+    }
+
+    *buf = 0; /* null terminate buffer */
+}
+
+void vector_init(struct vector *vector) {
```

Comment:
Could you move the vector code locally into ssh.c and ssh.h because it isn't being used outside of those files?